test/expect-tests/dune_console/dune_console_tests.ml

```diff
@@ -95,10 +120,13 @@ let%expect_test "Status line clearing with wrapping." =
   let module Console = New () in
   Console.Backend.set Console.Backend.progress;
   test_status_line_clearing_with_wrapping (module Console);
+  escape [%expect.output];
   [%expect
-    "\n\
-     This status line is a problem because of the fact that it is especially long and \
-     therefore will not be cleared \
-     properly.\r                                                                                                                        \
-     \rHere is another\r               \r "]
+    {| 
+This status line is a problem because of the fact that it is especially long
+and therefore will not be cleared properly.\r
+                                                                                                                        \r
+Here is another\r
+               \r
```

Comment:
@rgrinberg The most important part of the test is this. In our progress terminal, we have this logic for clearing the status line. This assumes that the status line is a single line since it does a return, prints space for the length of the line and then returns again. Doesn't work when there are multiple lines.

Since we are not in a dumb terminal, there is no reason for us to have this limitation in general. Therefore my fix will use the correct ansi control characters to clear lines (including those before) rather than having to calculate spaces. This logic might work on a dumb terminal, but the implementation there is to just print the status line and not clear it.